frontend/packages/dev-console/src/utils/useAddToProjectAccess.ts

```diff
@@ -36,6 +45,10 @@ export const useAddToProjectAccess = (activeNamespace: string): string[] => {
   const routeAccess = useAccessReview(resourceAttributes(RouteModel, activeNamespace));
   const serviceAccess = useAccessReview(resourceAttributes(ServiceModel, activeNamespace));
 
+  const serviceBindingEnabled = useSelector((state: RootState) =>
+    state.FLAGS.get(ALLOW_SERVICE_BINDING),
+  );
+
   React.useEffect(() => {
```

Comment:
This actually shouldn't be an effect. There's no reason for this hook to use `useState`.
`createResourceAccess` can easily be derived inside the hook function and at most we can replace `useEffect` with `useMemo` which returns `createResourceAccess` instead of setting state.